src/Microsoft.OData.Core/Json/TranscodingWriteStream.cs

```diff
@@ -0,0 +1,438 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="TranscodingWriteStream.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.OData.Json
+{
+    /// <summary>
+    /// A stream that allows converting from one encoding to another when writing to a stream.
+    /// </summary>
+    /// <remarks>
+    /// This is adapated from the .NET runtime's TranscodingStream available in .NET 5+

```

Comment:
> adapated 

Did we actually make any changes to this file?

Reply:
Yes. I removed stuff that weren't relevant. The .NET version includes both reading and writing APIs. I removed the write APIs. I also removed references to internal attributes and string resources used for error messages.